clang/lib/Sema/SemaExpr.cpp

```diff
@@ -7219,6 +7219,17 @@ Sema::ActOnCompoundLiteral(SourceLocation LParenLoc, ParsedType Ty,
   return BuildCompoundLiteralExpr(LParenLoc, TInfo, RParenLoc, InitExpr);
 }
 
+static bool IsInsideFunction(Scope *S) {
```

Comment:
```suggestion
static bool IsContainedByFunctionScope(const Scope *S) {
```
makes it more clear that we're checking whether the given scope is itself somewhere within a function scope.

Reply:
I can go either way. This is currently the first time we've needed this, so a static function makes sense. But if we need this a second time, then absolutely agreed it should be hoisted into `Scope`. WDYT?